Dockerfile

```diff
@@ -0,0 +1,20 @@
+FROM node:14-buster-slim
+
+WORKDIR /app
+
+COPY package.json package-lock.json ./
+
+RUN npm install --production
+
+COPY layouts layouts
+COPY lib lib
+COPY views views
+COPY sg/db.json sg/
+
+# TODO
+COPY config.json config.json
+COPY private-config.sample.json private-config.json
```

Comment:
How should we store the secrets? We'll be using a private docker registry so it won't be public, but I still think a better option would be to store secrets in environment variables, which are configured in DigitalOcean. That also means one doesn't need those credentials to deploy.

Reply:
The problem with environment variables is that, with the addition of Twitter credentials, we've grown an awful lot of them. Maybe we can just stuff a large JSON config into an environment variable??

Reply:
Yes, this is what I was intending. An `PRIVATE_CONFIG_JSON` variable set to the value of `cat private-config.json | base64`. This is how I've often handled groups of secrets in GitHub Actions and GCP.

Reply:
Do we know length limits? It's over 4000 now and can easily go over 9000 given enough new standards.

Reply:
I'm not able to tell from https://www.digitalocean.com/docs/app-platform/how-to/use-environment-variables/ or other linked documentation. I didn't realize we've reused this server to also tweet, but anyway we should have a setup that can deal with pretty large secrets. I don't have a solution I'm totally happy with yet.

Reply:
The Twitter stuff is most of this config so even if we split it out (and then we'd have to duplicate a bunch of surrounding infrastructure which makes me think it's not worth it) we'd still have this problem.

Reply:
I tried appending newlines to a real `private-config.json` until it was 1MB and base64-encoding that, resulting in a 2MB+ app.yaml file. Trying to deploy that results in a 500 Server Error, so clearly there are *some* limits we can't exceed here.

Reply:
50 kB was also too bit, but 10 kB works. So the limits are not super far away from what we might realistically use.